sdk/core/src/headers/mod.rs

```diff
@@ -65,6 +64,80 @@ impl Headers {
         self.0.get(&key.into())
     }
 
+    /// Get a header value or error if it is not found
+    pub fn get_or_err<T: Into<HeaderName>>(&self, key: T) -> crate::Result<&HeaderValue> {
+        let key: &HeaderName = &key.into();
+        let value = self.0.get(key);
+        value.ok_or_else(|| {
+            Error::with_message(ErrorKind::Other, || {
+                format!("header not found {}", key.as_str())
+            })
+        })
+    }
+
+    /// Get a header value as a str
+    pub fn get_as_str<T: Into<HeaderName>>(&self, key: T) -> Option<&str> {
+        self.get(key).map(|v| v.as_str())
+    }
```

Comment:
I think we should just get rid of the variants of this methods and only have the variants which return a `crate::Result`. If you need an option, you can just call `.ok()` on the `Result`. 

And of course, then we just use the simpler naming convention and not the `or_err` names.

Reply:
Sounds good. I added as #837.